src/ksyms.cpp

```diff
@@ -28,9 +42,67 @@ std::string Ksyms::resolve(uint64_t addr, bool show_offset)
   return symbol.str();
 }
 
-Ksyms::~Ksyms()
+#ifdef USE_BLAZESYM
+std::optional<std::string> Ksyms::resolve_blazesym_int(uint64_t addr,
+                                                       bool show_offset)
 {
-  if (ksyms_)
-    bcc_free_symcache(ksyms_, -1);
+  if (symbolizer_ == nullptr) {
+    symbolizer_ = blaze_symbolizer_new();
+    if (symbolizer_ == nullptr)
+      return std::nullopt;
+  }
```

Comment:
Note: if `blaze_symbolizer_new` fails for some reason, this will keep calling it for every symbolization request. I'm not sure how big of an issue that could be, just wanted to point it out.